ui/app/src/services/contentTypes.ts

```diff
@@ -431,6 +431,8 @@ function parseLegacyFormDefinition(definition: LegacyFormDefinition): ContentTyp
 				//   break;
 			}
 			dataSources[datasource.id].properties[property.name] = value;
+			// datasource also needs to be directly edited since it's set to dropTargetsLookup when datasource type is 'components'
+			datasource.properties[property.name] = value;
 		});
 		if (datasource.type === 'components') {
 			dropTargetsLookup[datasource.id] = datasource;
```

Comment:
We should avoid editing the item that was fetched or in general from the collection that's getting looped. Use the copy (see line 438ish).

Check if there's any implications of doing it this way.

Also, the comment is not clear. Improve pls.

```suggestion
			dropTargetsLookup[datasource.id] = dataSources[datasource.id];
```

Reply:
@rart The thing is that `datasources` if of type `DataSource`, and dropTargetsLookup is a lookup of LegacyDataSource with is used later to parse the validations. I created a copy of the legacy datasource and set it to dropTargetsLookup. Should we consider updating the dropTargetsLookup to be of type DataSource? (maybe as part of FE2 work)

Reply:
Yes. Pls file ticket.

Reply:
Add link to this ticket.

Reply:
https://github.com/craftercms/craftercms/issues/8216. Added TODO and link in code too.